charts/pulsar/templates/proxy/proxy-cluster.yaml

```diff
@@ -0,0 +1,230 @@
+    {{- end }}
+    {{- if .Values.proxy.nodeSelector }}
+    nodeSelector:
+{{ toYaml .Values.proxy.nodeSelector | indent 6 }}
+    {{- end }}
+    {{- if .Values.proxy.tolerations }}
+    tolerations:
+{{ toYaml .Values.proxy.tolerations | indent 6 }}
+    {{- end }}
+    terminationGracePeriodSeconds: {{ .Values.proxy.gracePeriod }}
+    jvmOptions:
+      extraOptions: []
+      memoryOptions:
+        - {{ .Values.proxy.configData.PULSAR_MEM }}
+      gcOptions:
+        - {{ .Values.proxy.configData.PULSAR_GC }}
+      gcLoggingOptions: []
+  config:
+    tls:
+      enabled: {{ and (.Values.tls.enabled) (.Values.tls.proxy.enabled) }}
+    custom:
+      zookeeperServers: "{{ template "pulsar.zookeeper.connect" . }}{{ .Values.metadataPrefix }}"
+      {{- if .Values.pulsar_metadata.configurationStore }}
+      configurationStoreServers: "{{ .Values.pulsar_metadata.configurationStore }}{{ .Values.pulsar_metadata.configurationStoreMetadataPrefix }}"
+      {{- end }}
+      {{- if not .Values.pulsar_metadata.configurationStore }}
+      configurationStoreServers: "{{ template "pulsar.zookeeper.connect" . }}{{ .Values.metadataPrefix }}"
+      {{- end }}
+{{ toYaml .Values.proxy.configData | indent 6 }}
+      {{- if .Values.auth.authentication.enabled }}
+      authenticationEnabled: "true"
+      authenticationProviders: "io.streamnative.pulsar.broker.authentication.AuthenticationProviderOIDCToken"
+      brokerClientAuthenticationPlugin: "org.apache.pulsar.client.impl.auth.AuthenticationToken"
+      PULSAR_PREFIX_vaultHost: {{ template "pulsar.vault.url" . }}
+      PULSAR_PREFIX_OIDCPublicKeyPath: "{{ template "pulsar.vault.url" . }}/v1/identity/oidc/.well-known/keys"
+      superUserRoles: "admin"
+      {{- end }}
+      {{- if .Values.auth.authorization.enabled }}
+      authorizationEnabled: "true"
+      authorizationProvider: "org.apache.pulsar.broker.authorization.PulsarAuthorizationProvider"
+      {{- end }}
+      {{- if and (.Values.tls.enabled) (.Values.tls.proxy.enabled) }}
+      {{- if .Values.certs.internal_issuer.enabled }}
+      tlsAllowInsecureConnection: "true"
+      tlsTrustCertsFilePath: "/etc/tls/pulsar-proxy-tls/ca.crt"
+      {{- end }}
+      {{- end }}
+{{ (.Files.Glob "conf/proxy/log4j2.yaml").AsConfig | indent 6 }}
+      clusterName: {{ template "pulsar.cluster" . }}
+      httpNumThreads: "8"
+      statusFilePath: "{{ template "pulsar.home" . }}/status"
+      # prometheus needs to access /metrics endpoint
+      webServicePort: "{{ .Values.proxy.ports.http }}"
+      {{- if or (not .Values.tls.enabled) (not .Values.tls.proxy.enabled) }}
+      servicePort: "{{ .Values.proxy.ports.pulsar }}"
+      {{- end }}
+      {{- if and .Values.tls.enabled .Values.tls.proxy.enabled }}
+      tlsEnabledInProxy: "true"
+      servicePortTls: "{{ .Values.proxy.ports.pulsarssl }}"
+      webServicePortTls: "{{ .Values.proxy.ports.https }}"
+      tlsCertificateFilePath: "/pulsar/certs/proxy/tls.crt"
```

Comment:
These cert paths might only be valid when adopting an existing proxy cluster?

Reply:
that's correct, updated